pkg/cloudproviders/aws/events/recorder.go

```diff
@@ -15,19 +15,52 @@ limitations under the License.
 package events
 
 import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/tools/record"
 )
 
-// Recorder is used to record events to the Kubernetes Event API
+type EventType byte
```

Comment:
If you make this a string, you get String() for free.

Reply:
Also, corev1 has 
```
// Valid values for event types (new types could be added in future)
const (
	// Information only and will not cause any problems
	EventTypeNormal string = "Normal"
	// These events are to warn that something might go wrong
	EventTypeWarning string = "Warning"
)
```

Reply:
This is the general way to do enums in Go, though. It is a more efficient memory representation of the type.